### PR TITLE
Reuse assistant code to create the mirror package

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes https://github.com/Lightning-AI/lightning/pull/15105/files#r1002933959

Also prunes a bunch of unused code

### Does your PR introduce any breaking changes? If yes, please list them.

None

cc @carmocca @akihironitta @borda

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -70,14 +70,16 @@ def _load_py_module(name: str, location: str) -> ModuleType:
 
 
 if __name__ == "__main__":
-    setup_tools = _load_py_module(name="setup_tools", location=os.path.join(".actions", "setup_tools.py"))
+    setup_tools = _load_py_module(name="setup_tools", location=os.path.join(_PATH_ROOT, ".actions", "setup_tools.py"))
+    assistant = _load_py_module(name="assistant", location=os.path.join(_PATH_ROOT, ".actions", "assistant.py"))
```
Comment (carmocca) [inline setup.py:74]: (not for this PR) I think these 2 files could be merged. Is there any advantage to having them separated?

**Review by akihironitta — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: .actions/assistant.py [hunk 2]
```diff
@@ -73,25 +37,74 @@ def _retrieve_files(directory: str, *ext: str) -> List[str]:
...
+    ...     "lightning_app",
+    ...     "delete_cloud_lightning_apps",
+    ...     "from lightning_app import",
+    ...     "lightning_apps = []",
+    ...     "lightning_app and pytorch_lightning are ours",
+    ...     "def _lightning_app():",
+    ...     ":class:`~lightning_app.core.flow.LightningFlow`"
+    ... ]
+    >>> mapping = [("lightning_app", "lightning.app"), ("pytorch_lightning", "lightning.pytorch")]
+    >>> _replace_imports(lns, mapping)  # doctest: +NORMALIZE_WHITESPACE
```
Comment (Borda) [inline .actions/assistant.py:53]: it is easier to read with pprint

File: .actions/assistant.py [hunk 3]
```diff
@@ -130,107 +143,17 @@ def replace_oldest_ver(requirement_fnames: Sequence[str] = REQUIREMENT_FILES_ALL
...
-
     @staticmethod
     def copy_replace_imports(
         source_dir: str, source_import: str, target_import: str, target_dir: Optional[str] = None
     ) -> None:
         """Recursively replace imports in given folder."""
-
         source_imports = source_import.strip().split(",")
         target_imports = target_import.strip().split(",")
-        assert len(source_imports) == len(target_imports), (
```
Comment (Borda) [inline .actions/assistant.py:201]: lets keep the assert

File: .actions/assistant.py [hunk 4]
```diff
@@ -130,107 +143,17 @@ def replace_oldest_ver(requirement_fnames: Sequence[str] = REQUIREMENT_FILES_ALL
...
-                    fp_new = fp
-
-                with open(fp_new, "w", encoding="utf-8") as fo:
-                    fo.writelines(py)
-            elif not fp.endswith(".pyc"):
-                fp_new = fp.replace(source_dir, target_dir)
-                os.makedirs(os.path.dirname(fp_new), exist_ok=True)
-                if os.path.abspath(fp) != os.path.abspath(fp_new):
-                    shutil.copy2(fp, fp_new)
+        copy_replace_imports(source_dir, source_imports, target_imports, target_dir=target_dir)
```
Comment (Borda) [inline .actions/assistant.py:153]: Does it call itself? I guess not but then lest call the function differently

File: .actions/assistant.py (hunk 2 above)
Comment (carmocca) [reply .actions/assistant.py:53]: We can add backlashes if you prefer, but the current break conforms to the line limit. pprint is not necessary

File: .actions/assistant.py (hunk 3 above)
Comment (carmocca) [reply .actions/assistant.py:201]: It's still there, in the other function

File: .actions/assistant.py (hunk 4 above)
Comment (carmocca) [reply .actions/assistant.py:153]: One is the assistant staticmethod, the other the function. They do the same so they should be called equally

**Review by Borda — APPROVED** [review]

**Pull request merged**